drivers/crypto/crypto_npcx_sha.c

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2022 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_sha
+
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/crypto/crypto.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sha_npcx, CONFIG_CRYPTO_LOG_LEVEL);
+
+#define NPCX_HASH_CAPS_SUPPORT	(CAP_SEPARATE_IO_BUFS | CAP_SYNC_OPS)
+#define NPCX_SHA256_HANDLE_SIZE 212
+#define NPCX_SHA_MAX_SESSION	1
+
+/* The status code returns from Nuvoton Cryptographic Library ROM APIs */
+enum ncl_status {
+	NCL_STATUS_OK = 0xA5A5,
```

Comment:
Unrelated to this PR, but is there a bug in the [chromium EC driver](https://source.chromium.org/chromiumos/chromiumos/codesearch/+/HEAD:src/platform/ec/chip/npcx/sha256_chip.c;l=14)?  That driver uses `NCL_STATUS_OK = 0`.

Reply:
Indeed. `NCL_STATUS_OK = 0` is not correct in the Cros-EC driver. I will submit a CL to fixe it.